prov/shm/src/smr_ep.c

```diff
@@ -1329,6 +1329,8 @@ static int smr_ep_ctrl(struct fid *fid, int command, void *arg)
 			}
 		}
 
+		ep->smr_async_device_copy = rocr_is_ipc_enabled();
```

Comment:
`rocr_is_ipc_enabled` is internal to the HMEM API. Might be better to check `hmem_ops[FI_HMEM_ROCR].initialized`

Reply:
The call should be ofi_hmem_is_ipc_enabled().  This may need to be called in a loop for all HMEM, and if any of them return true, then we need to call progress_ipc_list() below.

Reply:
ROCR is currently the only iface that uses the async protocol so I think they are trying to optimize this out for CUDA which doesn't need it.
@sunkuamzn @a-szegel Are there plans or desires to extend the CUDA protocol to use it asynchronously? If so, I agree with Sean, make it general to all interfaces. If the plan is to always use CUDA synchronously and you're looking to optimize that case specifically, then I think it's ok to call it directly. If the aim here is only to optimize the non HMEM case, then I would still make it general.

Reply:
We don't have plans to implement async CUDA copies in the near future. We need to do some benchmarking to see if it's even worth doing.

Reply:
kept at ROCR IPC and added a comment.